libs/core/langchain_core/tools.py

```diff
@@ -371,46 +374,25 @@ def get_input_schema(
 
     def invoke(
         self,
-        input: Union[str, Dict],
+        input: Union[str, Dict, ToolCall],
         config: Optional[RunnableConfig] = None,
         **kwargs: Any,
     ) -> Any:
-        config = ensure_config(config)
-        return self.run(
-            input,
-            callbacks=config.get("callbacks"),
-            tags=config.get("tags"),
-            metadata=config.get("metadata"),
-            run_name=config.get("run_name"),
-            run_id=config.pop("run_id", None),
-            config=config,
-            **kwargs,
-        )
+        tool_input, kwargs = _prep_run_args(input, config, **kwargs)
+        return self.run(tool_input, **kwargs)
 
     async def ainvoke(
         self,
-        input: Union[str, Dict],
+        input: Union[str, Dict, ToolCall],
         config: Optional[RunnableConfig] = None,
         **kwargs: Any,
     ) -> Any:
-        config = ensure_config(config)
-        return await self.arun(
-            input,
-            callbacks=config.get("callbacks"),
-            tags=config.get("tags"),
-            metadata=config.get("metadata"),
-            run_name=config.get("run_name"),
-            run_id=config.pop("run_id", None),
-            config=config,
-            **kwargs,
-        )
+        tool_input, kwargs = _prep_run_args(input, config, **kwargs)
+        return await self.arun(tool_input, **kwargs)
 
     # --- Tool ---
 
-    def _parse_input(
-        self,
-        tool_input: Union[str, Dict],
-    ) -> Union[str, Dict[str, Any]]:
+    def _parse_input(self, tool_input: Union[str, Dict]) -> Union[str, Dict[str, Any]]:
```

Comment:
What can tool input be here? (if you know of the top of you head and have time could be good to update the doc-string)